### DISCUSSION TITLE
eth_subscribe for logs

### DISCUSSION THREAD

**Discussion by pinebit** [erigontech/erigon#4946] [General]
Hello!

We are trying to integrate Erigon and Chainlink, and we hit the missing feature of subscribing to logs mentioned here: https://github.com/ledgerwatch/erigon/issues/4030#issuecomment-1113964017

We were going to build a workaround such that on every new head we would simply request logs for the block, however after digging into this further, we found one edge case described here: https://geth.ethereum.org/docs/rpc/pubsub

> In case of a chain reorganization previous sent logs that are on the old chain will be resend with the removed property set to true. Logs from transactions that ended up in the new chain are emitted. Therefore a subscription can emit logs for the same transaction multiple times.

This says, that in case of re-org the removed logs will be re-delivered. However, having the single subscription for heads we are missing this important feature. The implementation would become extremely difficult if we have to detect reorgs with heads and then backfilling past logs that might be removed.

Do you have a better workaround for the missing logs subscription?

Thank you!

**Comment by enriavil1** [comment]
There is currently a `subscribeFilterLogs` inside of Erigon which should work like a subscribe log

**Reply by enriavil1** [reply]
I am not sure wether it is functional or not it seems like it can return new data, but it won't return past data.
